Fixes#1

bodyankush opened the pull request:
Правки:
1) Убрал в коде shoulBe(visible). 
2) Поставил более короткие paths.
3) Убрал не нужные методы в LoginElements классе, обновил Тест NeomUserCanBeSuccessfullyLoggedIn.
4) Добавил к нвазаниям “…Pages”.
5) ? NewDigitalIdRequest має включати степпер - не понял
6) Переименовал SiteTree в Pages.
7) Переместил Pages в main.
8) Создал App с инстансами страниц, усешно импортировал в тест.
9) Поставил Логин в @BeforeEach
10) ? Не получилось с методом hasOptions()

BodyanQA committed:
Fixes#1

Igur007 commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
create .gitignore file in the root 
add .idea folder to it
such files should never appear in repo

Igur007 commented on src/main/java/Pages/Services/ServicesPage.java:
```diff
@@ -1,9 +1,9 @@
-package Web.SiteTree.Services;
+package Pages.Services;
```
all classes that describe pages, elements, helpers should be located in folders under src/main/java
remove them from src/test/java and keep only test files there

bodyankush replied:
Fixed.

Igur007 commented on src/main/java/Pages/App/App.java:
```diff
@@ -0,0 +1,20 @@
+        services = new Services();
+        internalServices = new InternalServices();
+        newDigitalIdRequest = new NewDigitalIdRequest();
+        idTypeStep = new IdTypeStep();
```
idTypeStep = new IdTypeStep(); should be created not here but inside of NewDigitalIdRequest
this stepper does not exist outside of creating new digital id

Igur007 replied:
same valid for instance of InternalServices, it's a part of new Services() 

later your usage would be smth like 
app.services.internalServices.newDigitalId.idTypeStep ....

so think about it as it's developed and it looks like in your UI. e.g. if you can not deal with internalServices until you go to Services, then it has be inside of Services. The same logic for any other page/component

bodyankush replied:
Done.

Igur007 commented on src/main/java/Pages/App/IdTypeStep.java:
```diff
@@ -0,0 +1,6 @@
+
+import Pages.Services.InternalServices.DigitalIdRequests.NewId.Steper.IdType.IdTypeStepPage;
+
+public class IdTypeStep extends IdTypeStepPage {
```
Step class should not extend the page class.
Step is just a separate class that describes specific part of the page, so the instance of IdTypeStep should be created in constructor of IdTypeStepPage

same is valid for other step classes

bodyankush replied:
done

Igur007 commented on src/main/java/Pages/Services/InternalServices/DigitalIdRequests/NewId/Steper/IdType/IdTypeStepPage.java:
```diff
@@ -0,0 +1,21 @@
+
+public class IdTypeStepPage {
+    public final SelenideElement stepNaming = $x("//div[contains(@class, \"title\") and text()=\"ID type\"]");
+    public final SelenideElement titleNaming = $x("//*[@id=\"digitalIdForm\"]/div[1]/div/h3");
```
/div[1]/div/h3 - bad locator

Igur007 commented on src/main/java/Pages/Services/InternalServices/DigitalIdRequests/NewId/Steper/IdType/IdTypeStepPage.java:
```diff
@@ -0,0 +1,21 @@
+public class IdTypeStepPage {
+    public final SelenideElement stepNaming = $x("//div[contains(@class, \"title\") and text()=\"ID type\"]");
+    public final SelenideElement titleNaming = $x("//*[@id=\"digitalIdForm\"]/div[1]/div/h3");
+    public final SelenideElement subTitleNaming = $x("//*[@id=\"digitalIdForm\"]/div[1]/div/p");
```
/div[1]/div/p - bad locator

Igur007 commented on src/main/java/Pages/Services/InternalServices/DigitalIdRequests/NewId/Steper/IdType/IdTypeStepPage.java:
```diff
@@ -0,0 +1,21 @@
+    public final SelenideElement titleNaming = $x("//*[@id=\"digitalIdForm\"]/div[1]/div/h3");
+    public final SelenideElement subTitleNaming = $x("//*[@id=\"digitalIdForm\"]/div[1]/div/p");
+    public final SelenideElement idTypeDropdown = $x("//div[@role='combobox' and contains(@class, 'ng-input')]");
+    public final SelenideElement backButton = $x("//*[@id=\"digitalIdForm\"]/div[3]/div[1]/button");
```
/div[3]/div[1]/button bad locator

Igur007 commented on src/main/java/Pages/Services/InternalServices/DigitalIdRequests/NewId/Steper/IdType/IdTypeStepPage.java:
```diff
@@ -0,0 +1,21 @@
+    public final SelenideElement subTitleNaming = $x("//*[@id=\"digitalIdForm\"]/div[1]/div/p");
+    public final SelenideElement idTypeDropdown = $x("//div[@role='combobox' and contains(@class, 'ng-input')]");
+    public final SelenideElement backButton = $x("//*[@id=\"digitalIdForm\"]/div[3]/div[1]/button");
+    public final SelenideElement nextButton = $x("//*[@id=\"digitalIdForm\"]/div[3]/div[2]/button");
```
bad locator /div[3]/div[1]/button

Igur007 commented on src/test/java/Web/TestCases/NewDigitalId/ExpectedElementsArePresentAtStepIdType.java:
```diff
@@ -1,65 +1,49 @@
 package Web.TestCases.NewDigitalId;
```
TestCases in unnecessary folder, get rid of it

Igur007 commented on src/test/java/Web/TestCases/NewDigitalId/ExpectedElementsArePresentAtStepIdType.java:
```diff
@@ -1,65 +1,49 @@
 public class ExpectedElementsArePresentAtStepIdType extends NeomUserCanBeSuccessfullyLoggedIn {
-    @Test
-    public void expectedElementsArePresentAtStepIdType(){
+    private App app;
```
Test class better to start with Test word

bodyankush replied:
Done

Igur007 commented on src/test/java/Web/TestCases/NewDigitalId/ExpectedElementsArePresentAtStepIdType.java:
```diff
@@ -1,65 +1,49 @@
-    public void expectedElementsArePresentAtStepIdType(){
+    private App app;
+    @BeforeMethod
+    public void setUp(){
```
this is common setup, so you have to create a base test class, keep this common setup there
your test class should extend base test class and inherit it

Igur007 commented on src/test/java/Web/TestCases/NewDigitalId/ExpectedElementsArePresentAtStepIdType.java:
```diff
@@ -1,65 +1,49 @@
+        app.idTypeStep.nextButton.shouldBe(disabled);
     }
     @Test
     public void expectedValuesArePresentAtIdTypeDropdown(){
```
I don't see a test logic where the stepper form is filled in, user clicks Next and appears on the next stepper. And this test has to be the critical one, as it's about logic

BodyanQA committed:
add .idea to gitignore

BodyanQA committed:
Removed LoginElements from the Login Test

BodyanQA committed:
idTypeStep = new IdTypeStep(); should be created not here but inside of NewDigitalIdRequest
this stepper does not exist outside of creating new digital id
Fixed all instances by the following example

BodyanQA committed:
removed TestCases folder

BodyanQA committed:
Add "Test" in the beginning of the Test classes, also removed empty Test classes which i created in adavnce

BodyanQA committed:
Created instance of IdTypeStep in constructor of IdTypeStepPage

BodyanQA committed:
removed code added accidently
